modules/publinkIdSystem.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * This module adds the PublinkId to the User ID module
+ * The {@link module:modules/userId} module is required
+ * @module modules/conversantIdSystem
+ * @requires module:modules/userId
+ */
+
+import {submodule} from '../src/hook.js';
+import {getStorageManager} from '../src/storageManager.js';
+import {ajax} from '../src/ajax.js';
+import * as utils from '../src/utils.js';
+import {uspDataHandler} from '../src/adapterManager.js';
+
+const MODULE_NAME = 'publink';
+const GVLID = 24;
+const SERVER_URL = utils.parseUrl('https://proc.ad.cpe.dotomi.com/cvx/client/direct/publink');
+
+export const storage = getStorageManager(GVLID);
+
+function publinkIdUrl(params, consentData) {
+  let url = {...SERVER_URL};
+  url.search = {
+    deh: params.e,
+    mpn: 'Prebid.js',
+    mpv: '$prebid.version$',
+  };
+  if (consentData) {
+    url.search.gdpr = (consentData.gdprApplies) ? 1 : 0;
+    url.search.gdpr_consent = consentData.consentString;
+  }
+
+  const usPrivacyString = uspDataHandler.getConsentData();
+  if (usPrivacyString && typeof usPrivacyString === 'string') {
+    url.search.us_privacy = usPrivacyString;
+  }
+
+  return utils.buildUrl(url);
+}
+
+function makeCallback(config = {}, consentData) {
+  return function(prebidCallback) {
+    const options = {method: 'GET', withCredentials: true};
+    let handleResponse = function(responseText, xhr) {
+      let response = JSON.parse(responseText);
```

Comment:
This should be done inside the block at line 45.  Check the status first before parsing.